tuf/api/metadata.py

```diff
@@ -154,96 +122,166 @@ def verify(self) -> bool:
                         continue
                     else:
                         # Avoid https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-6174
-                        verified_keyids |= keyid
+                        verified_keyids.add(keyid)
+
                         break
 
-        return len(verified_keyids) >= self.__keyring.threshold.least
+        return len(verified_keyids) >= key_ring.threshold.least
 
-    def write_to_json(self, filename: str, storage_backend: StorageBackendInterface = None) -> None:
+    def write_to_json(
+            self, filename: str,
+            storage_backend: StorageBackendInterface = None) -> None:
          with tempfile.TemporaryFile() as f:
             f.write(_get_written_metadata(self.sign()).encode_canonical())
             persist_temp_file(f, filename, storage_backend)
 
-class Timestamp(Metadata):
-    def __init__(self, consistent_snapshot: bool = True, expiration: datetime = datetime.today(), keyring: KeyRing = None, version: int = 1) -> None:
-        super().__init__(consistent_snapshot, expiration, keyring, version)
-        self.snapshot_fileinfo = {}
 
-    @classmethod
-    def read_from_json(cls, filename: str) -> Metadata:
-        md = Metadata.read_from_json(filename)
-        timestamp = cls(md.consistent_snapshot, md.expiration, md.keyring, md.version)
-        timestamp.snapshot_fileinfo = md._signed['meta']
-        tuf.formats.TIMESTAMP_SCHEMA.check_match(timestamp.signed)
-        timestamp._signatures = md._signatures
-        return timestamp
+class Signed:
+    # NOTE: Signed is a stupid name, because this might not be signed yet, but
+    # we keep it to match spec terminology (I often refer to this as "payload",
+    # or "inner metadata")
+
+    # TODO: Re-think default values. It might be better to pass some things
+    # as args and not es kwargs. Then we'd need to pop those from
+    # signable["signed"] in read_from_json and pass them explicitly, which
+    # some say is better than implicit. :)
+    def __init__(
+            self, _type: str = None, version: int = 0,
+            spec_version: str = None, expires: datetime = None
+        ) -> None:
+        # TODO: How much init magic do we want?
+
+        self._type = _type
+        self.spec_version = spec_version
+
+        # We always intend times to be UTC
+        # NOTE: we could do this with datetime.fromisoformat() but that is not
+        # available in Python 2.7's datetime
+        # NOTE: Store as datetime object for convenient handling, use 'expires'
+        # property to get the TUF metadata format representation
+        self.__expiration = iso8601.parse_date(expires).replace(tzinfo=None)
+
+        if version < 0:
+            raise ValueError(f'version must be < 0, got {version}')
```

Comment:
this error message doesn't match the conditional

Reply:
Good catch. Let's keep the note around and I'll fix it together with other validation updates. What do you think about the related ideas I floated in the doc header todo list?

Reply:
I'll comment on the ideas you floated in-line 😃